Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -145,8 +145,33 @@ public struct HTTPResponseHead: Equatable {
     }
 }
 
-fileprivate typealias HTTPHeadersStorage = [String: [(String, String)]] // [lowerCasedName: [(originalCaseName, value)]
+/// The Index for a header name or value that points into the underlying `ByteBuffer`.
+struct HTTPHeaderIndex {
```

Comment:
nit: i think it would be less confusing to read this code (and probably less error prone to modify it) if there was another struct that contained two of these: one for the name and one for the value.

Reply:
@tanner0101 so you suggest having an extra struct which holds two of these structs and store the extra struct in the array (which means removing the % 2 logic) ?

Reply:
@tanner0101 done in https://github.com/apple/swift-nio/commit/f533f4d193a3db6f5c958db6ba9d638bb2dcf887.... also gave a nice speed up due less array accesses  ;)